core/src/com/unciv/ui/tilegroups/TileSetStrings.kt

```diff
@@ -66,4 +72,30 @@ class TileSetStrings {
         if (baseTerrain != null) return getString(tilesLocation, baseTerrain, "+", city)
         else return cityTile
     }
+
+    /** Fallback [TileSetStrings] to use when the currently chosen tileset is missing an image. */
+    val fallback by lazy {
+        if (fallbackDepth <= 0 || tileSetConfig.fallbackTileSet == null)
+            null
+        else
+            TileSetStrings(tileSetConfig.fallbackTileSet!!, fallbackDepth-1)
+    }
+    /**
+     * @param image An image path string, such as returned from an instance of [TileSetStrings].
+     * @param fallbackImage A lambda function that will be run with the [fallback] as its receiver if the original image does not exist according to [ImageGetter.imageExists].
+     * @return The original image path string if its image exists, or the return result of the [fallbackImage] lambda if the original image does not exist.
+     * */
+    fun orFallback(image: String, fallbackImage: TileSetStrings.() -> String): String {
+        return if (fallback == null || ImageGetter.imageExists(image))
+            image
+        else
+            fallback!!.run(fallbackImage)
+    }
+    /** @see orFallback */
+    fun orFallback(image: TileSetStrings.() -> String, fallbackImage: TileSetStrings.() -> String)
+            = orFallback(this.run(image), fallbackImage)
```

Comment:
I wonder if using extension function syntax (`fallback!!.fallbackImage()`, `this.image()`) might allow more compiler optimizations. But `.run()` should make it fairly immune to name collisions.